doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -91,72 +79,87 @@ and tables:
+      is another topic for discussion/action.
 
+   .. exn:: There is no option @setting_name.
 
-.. cmd:: Set @option {| @num | @string }
-   :name: Set @option
+      This message also appears for unknown flags.
 
-   Sets :token:`option` to the specified value.
+.. cmd:: Set @setting_name {| @int | @string }
+   :name: Set option
 
-.. cmd:: Unset @option
-   :name: Unset @option
+   Sets the option :n:`@setting_name` to the specified value, which
+   must be of the appropriate type for the option.
 
-   Sets :token:`option` to its default value.
+   This command supports the :attr:`local`, :attr:`global` and :attr:`export` attributes.
+   They are described :ref:`here <set_unset_scope_qualifiers>`.
 
-.. cmd:: Test @option
+.. cmd:: Unset @setting_name
+   :name: Unset
 
-   Prints the current value of :token:`option`.
+   If :n:`@setting_name` is a flag, it is set to off.  If :n:`@setting_name` is an option, it is
+   set to its default value.
 
-.. cmd:: Print Options
+   This command supports the :attr:`local`, :attr:`global` and :attr:`export` attributes.
+   They are described :ref:`here <set_unset_scope_qualifiers>`.
 
-   Prints the current value of all flags and options, and the names of all tables.
+   .. todo doesn't work for "Diffs"
+
+.. cmd:: Test @setting_name
+   :name: Test flag/option
+
+   Prints the current value of the flag or option :n:`@setting_name`.
 
+.. cmd:: Print Options
 
-.. cmd:: Add @table {| @string | @qualid }
-   :name: Add @table
+   Prints the current value of all flags and options, and the names of all tables.
 
-   Adds the specified value to :token:`table`.
 
-.. cmd:: Remove @table {| @string | @qualid }
-   :name: Remove @table
+.. cmd:: Add @setting_name {+ {| @qualid | @string } }
+   :name: Add table
 
-   Removes the specified value from :token:`table`.
+   Adds the specified values to the table :n:`@setting_name`.
 
-.. cmd:: Test @table for {| @string | @qualid }
-   :name: Test @table for
+.. cmd:: Remove @setting_name {+ {| @qualid | @string } }
+   :name: Remove table
 
-   Reports whether :token:`table` contains the specified value.
+   Removes the specified value from the table :n:`@setting_name`.
 
-.. cmd:: Print Table @table
-   :name: Print Table @table
+.. cmd:: Test @setting_name {? for {+ {| @qualid | @string } } }
+   :name: Test table
 
-   Prints the values in :token:`table`.
+   If a value is given, reports whether the table :n:`@setting_name` contains the specified value.
+   Otherwise, this is equivalent to :cmd:`Print Table`.
 
-.. cmd:: Test @table
+.. cmd:: Print Table @setting_name
 
-   A synonym for :cmd:`Print Table @table`.
+   Prints the values in the table :n:`@setting_name`.
 
 .. cmd:: Print Tables
 
    A synonym for :cmd:`Print Options`.
 
-Locality attributes supported by :cmd:`Set` and :cmd:`Unset`
+.. _set_unset_scope_qualifiers:
+
+Locality attributes supported by :n:`Set` and :cmd:`Unset`
```

Comment:
Keep the `cmd` please. Like I said, I think the index for the command should be just `Set`. If you want to consolidate the two variants in one, that's fine by me.